isotovideo

```diff
@@ -302,7 +303,7 @@ if ($bmwqemu::vars{_EXIT_AFTER_SCHEDULE}) {
 testapi::init();
 
 # make sure the needles are initialized
-$bmwqemu::vars{NEEDLES_GIT_HASH} = OpenQA::Isotovideo::Utils::calculate_git_hash(needle::init);
+$bmwqemu::vars{NEEDLES_GIT_HASH} = checkout_git_refspec(needle::init => 'NEEDLES_GIT_REFSPEC');
```

Comment:
This initializes the needle directory and then switches the branch. It would make more sense to do this in the opposite order. To change the order you likely need to split `needle::init` into `needle::determine_needle_dir` and `needle::scan_needle_dir`.

Reply:
I have a better idea. I just moved the part *within* needle::init. This also makes isotovideo look nicer and cleaner as we have `needle::init` just after `testapi::init` :)